runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/spring/ServiceAnnotationBeanPostProcessor.java

```diff
@@ -64,6 +64,25 @@
     private BindingConverterFactory bindingConverterFactory;
     private ApplicationContext      applicationContext;
 
+    /**
+     * To construct a ServiceAnnotationBeanPostProcessor via a Spring Bean
+     * sofaRuntimeContext and sofaRuntimeProperties will be obtained from applicationContext
+     * @param bindingAdapterFactory
+     * @param bindingConverterFactory
+     */
+    public ServiceAnnotationBeanPostProcessor(BindingAdapterFactory bindingAdapterFactory,
+                                              BindingConverterFactory bindingConverterFactory) {
+        this.bindingAdapterFactory = bindingAdapterFactory;
+        this.bindingConverterFactory = bindingConverterFactory;
+    }
+
+    /**
+     * To manually construct a ServiceAnnotationBeanPostProcessor
+     * @param sofaRuntimeContext
+     * @param sofaRuntimeProperties
+     * @param bindingAdapterFactory
+     * @param bindingConverterFactory
+     */
     public ServiceAnnotationBeanPostProcessor(SofaRuntimeContext sofaRuntimeContext,
```

Comment:
Can we delete this constructor and mock the application context in tests?

Reply:
yes, it can be deleted.